roles/scm/git/tasks/pull.yml

```diff
@@ -0,0 +1,31 @@
+---
+
+- name: Prepare SCM credentials
+  include_tasks: prep.yml
+
+- name: Create temporary directory for the repository
+  tempfile:
+    state: directory
+    prefix: "repo-"
+  register: result_scm_dir
+  changed_when: false
+  when:
+    - not repository.path is defined
+
+- name: Set directory for the repository
+  set_fact:
+    scm_dir: "{{ repository.path if repository.path is defined else result_scm_dir.path }}"
+
+- name: "Clone repository to temporary location"
+  git:
+    clone: true
+    repo: "{{ scm_repository }}"
+    dest: "{{ scm_dir }}"
+    force: "{{ repository.force | default(omit) }}"
+    key_file: "{{ ssh_key | default(omit) }}"
+    remote: "{{ repository.remote | default('origin') }}"
+    update: "{{ repository[update] | default(omit) }}"
```

Comment:
why square brackets? If it's a special case, please make sure to comment inline

Reply:
Yes it's a special case due to reserved word, will update comments to suit.